extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/FullyFeaturedServerJacksonMessageBodyWriter.java

```diff
@@ -55,23 +62,56 @@ public void writeResponse(Object o, Type genericType, ServerRequestContext conte
                 var customSerializationValue = ResteasyReactiveServerJacksonRecorder.customSerializationForMethod(methodId);
                 if (customSerializationValue != null) {
                     ObjectWriter objectWriter = perMethodWriter.computeIfAbsent(methodId,
-                            new MethodObjectWriterFunction(customSerializationValue, genericType, originalMapper));
+                            new MethodObjectWriterFunction(customSerializationValue, genericType, effectiveMapper));
                     objectWriter.writeValue(stream, o);
                     return;
                 }
 
                 Class<?> jsonViewValue = ResteasyReactiveServerJacksonRecorder.jsonViewForMethod(methodId);
                 if (jsonViewValue != null) {
-                    defaultWriter.withView(jsonViewValue).writeValue(stream, o);
+                    getEffectiveWriter(effectiveMapper).withView(jsonViewValue).writeValue(stream, o);
                     return;
                 }
             }
-            defaultWriter.writeValue(stream, o);
+            getEffectiveWriter(effectiveMapper).writeValue(stream, o);
         }
         // we don't use try-with-resources because that results in writing to the http output without the exception mapping coming into play
         stream.close();
     }
 
+    private ObjectWriter getEffectiveWriter(ObjectMapper effectiveMapper) {
+        if (effectiveMapper == originalMapper) {
+            return defaultWriter;
+        }
+        return contextResolverMap.computeIfAbsent(effectiveMapper, new Function<>() {
+            @Override
+            public ObjectWriter apply(ObjectMapper objectMapper) {
+                return createDefaultWriter(effectiveMapper);
+            }
+        });
+    }
+
+    /**
+     * Obtains the user configured {@link ObjectMapper} if there is a {@link ContextResolver} configured.
+     * Otherwise, returns the default {@link ObjectMapper}.
+     */
+    private ObjectMapper getEffectiveMapper(Object o, ServerRequestContext context) {
+        ObjectMapper effectiveMapper = originalMapper;
+        ContextResolver<ObjectMapper> contextResolver = providers.getContextResolver(ObjectMapper.class,
+                context.getResponseMediaType());
```

Comment:
It's more of a general question but do you handle `@ConstrainedTo` for providers in RESTEasy Reactive? (to constrain providers to server/client)

Reply:
That's an excellent question, unfortunately this is a limitation of the (pretty bad) JAX-RS `Provider` API. If you look at `javax.ws.rs.ext.Providers`, there is way to limit the results using the `javax.ws.rs.RuntimeType`

Reply:
But RESTEasy Classic implements it, I'm sure because I used this feature already.

Reply:
Implements what?

Reply:
Support for `@ConstrainedTo`. But maybe I misunderstood what you were saying.

Note that it doesn't mean you can't merge this PR, it's just a question that came to mind.

Reply:
I don't see `@ConstrainedTo` being taken into account anywhere in RESTEasy Classic. Do you have an example suggesting otherwise?